dao/endpoint_dao.go

```diff
@@ -84,11 +85,14 @@ func (a *EndpointDaoImpl) Tenant() *int64 {
 }
 
 func (a *EndpointDaoImpl) CanEndpointBeSetAsDefaultForSource(sourceId int64) bool {
-	endpoint := &m.Endpoint{}
+	var count int64
 	// add double quotes to the "default" column to avoid any clashes with postgres' "default" keyword
-	result := DB.Where(`"default" = true AND source_id = ?`, sourceId).First(&endpoint)
+	result := DB.Model(&m.Endpoint{}).Where(`"default" = true AND source_id = ?`, sourceId).Count(&count)
+	if result.Error != nil {
+		logging.Log.Errorf("cannot check whether endpoint can be set %s", result.Error.Error())
+	}
 
-	return result.Error == nil
```

Comment:
Regarding this, wouldn't changing the condition to `result.Error != nil` do the job? I'm thinking that the way I wrote it, it was wrong because it was doing the following:

| Default endpoints in database | `.Find` returns result | `result.Error` is `nil` | `CanEndpointBeSetAsDefaultForSource`? | 
|-------------------------------|----------------------|---------------------|-------------------------------------|
| yes                           | yes                  | yes                  | true                                |
| no                            | no                   | no                 | false                               |

... when it should return the opposite thing. `true` when there are no default endpoints, and `false` when there is at least one. `.Count` might be easier to follow though...

Reply:
I'd also add that `.First()` does a `LIMIT 1` on the query vs `.Count()` doing a full table scan. I generally lean towards `.First()` and comparing the error instead.

@MikelAlejoBR I think you're right - we would need to inverse it.

Reply:
yes, that's right (also LIMIT 1 👍 ) I put it back with `result.Error != nil` . 


**NOT for this PR - my thoughts about it:**
```
func (a *EndpointDaoImpl) CanEndpointBeSetAsDefaultForSource(sourceId int64) bool {
	endpoint := &m.Endpoint{}

	// add double quotes to the "default" column to avoid any clashes with postgres' "default" keyword
	result := DB.Where(`"default" = true AND source_id = ?`, sourceId).First(&endpoint)
	return result.Error != nil
}
```

as I am looking on code - it is harder to understand as and it is just checking existence of record.
Maybe some refactoring to extract method about "query exists" and I believe that it could be used also in other places.

Reply:
You are right, the naming of the function makes it hard to reason about — in fact I think that's why I got that return value the other way around. I think your idea of renaming it to `RecordExists`, `DefaultEndpointExists` or any other name will make it easier to understand :+1: